qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -280,6 +280,21 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         return self._run(dag, tracker)
 
     def _run(self, dag: DAGCircuit, tracker: QubitTracker) -> DAGCircuit:
+        # Check if HighLevelSynthesis can be skipped.
+        all_skipped = True
+        for node in dag.topological_op_nodes():
+            qubits = tuple(dag.find_bit(q).index for q in node.qargs)
+            if not (
+                dag.has_calibration_for(node)
+                or len(node.qargs) < self._min_qubits
+                or node.is_directive()
+                or self._definitely_skip_node(node, qubits)
+            ):
```

Comment:
This set of checks is the exact same set that we do during the loop below - should they be folded into `_definitely_skip_node` so it's shorter to read and everything stays in sync?

Looking quickly, it seems like the internal logic of HLS has changed slightly since I wrote that fast path within the loop, so the logic around having a separate `_definitely_skip_node` might be a bit different to when I first wrote it.

Reply:
Good point, done in [0962be4](https://github.com/Qiskit/qiskit/pull/13070/commits/0962be43f9f3fef986f04aa62a7c3a3ba6b955d6).